src/utils/animatedInterpolate.js

```diff
@@ -0,0 +1,30 @@
+/**
+ * get the first and last animated interpolation
+ * When the return value is null, the scene should be set to invisible.
+ * @returns {{first: Number, last:Number} || null}
+ */
+function animatedInterpolate({ scene, scenes }) {
```

Comment:
hi @94cstyles - I'm sorry for delivering the review in chunks but I haven't had time to sit and do a proper review until now. I'd like to ask you again for some minor changes, only in this one file -
1 . please rename this function so it is very clear about what it does. `animatedInterpolate` doesn't convey much information.
2 . please rename the variables used in the body of this function. Right now there are variables `activeIndex`, `scenesActiveIndex`, `scenesCurrentIndex`. Some of these adjectives / names sound very similar (active, current, `activeIndex`, `scenesActiveIndex`) and people who read the code will not understand the difference (eg. `activeIndex` vs. `scenesActiveIndex` vs. `scenesCurrentIndex`). Could you change the names so that the reader doesn't get so easily confused?
3 . please add flow types

the body of the function could also be made nicer, by not mutating the `first` and `last` variables and rewriting the if, for example

```js
function getFirstAndLastSceneIndicesForInterpolationInputRange({ scene, scenes }) {
  const lastIndex = scenes.length - 1;
  const index = scene.index;
  const isBack = !scenes[lastIndex].isActive;
  if (!isBack) {
    return { first: index - 1, last: index + 1 };
  } else {
    // please rename the constants; you can even put this in a separate function if you like
    const activeScene = scenes.find(item => item.isActive);
    const scenesActiveIndex = scenes.findIndex(item => item === activeScene);
    const activeIndex = activeScene.index;
    const scenesCurrentIndex = scenes.findIndex(item => item === scene);

    if (scenesCurrentIndex === lastIndex) {
      return { first: activeIndex, last: index + 1 };
    } else if (index === activeIndex && scenesCurrentIndex === scenesActiveIndex) { // eslint-disable-line
      return { first: index - 1, last: lastIndex };
    } else if (scenesCurrentIndex > scenesActiveIndex || index === activeIndex) { // eslint-disable-line
      return null;
    }
  }
}
```

That's it! Thank you so much much for your work. As a side note - it seems like you've been using force push to keep all your work in a single commit. Please do not do this - it makes the review process harder by not seeing the changes you made after addressing the review requests. We can squash the commits into one before merging to the master. Thank you.

Reply:
Thank you for your feedback, let me realize that my lack.